app/controllers/products_controller.rb

```diff
@@ -0,0 +1,56 @@
+class ProductsController < ApplicationController
+  def index
+    @vendor_id = params[:vendor_id]
+    @id = params[:id]
+    @products = Vendor.find(@vendor_id).products
+  end
+
+  def new
+    @product = Product.new
+    @url_path = vendor_products_path
+  end
+
+  def create
+    vendor_id = params[:vendor_id]
+    Product.create(
+      name: product_params[:product][:name],
+      vendor_id: session[:vendor_id])
+    redirect_to "/vendors/#{vendor_id}/products"
```

Comment:
It is better to use a named path rather than the URL itself. Is there a named path for this route that you could use instead?